setup.py

```diff
@@ -0,0 +1,104 @@
+# Copyright (c) 2016-2018, The University of Texas at Austin
+# & University of California, Merced.
+#
+# All Rights reserved.
+# See file COPYRIGHT for details.
+#
+# This file is part of the hIPPYlib library. For more information and source code
+# availability see https://hippylib.github.io.
+#
+# hIPPYlib is free software; you can redistribute it and/or modify it under the
+# terms of the GNU General Public License (as published by the Free
+# Software Foundation) version 2.0 dated June 1991.
+
+
+from setuptools import setup, find_packages
+from os import path
+from io import open
+import re
+
+
+try:
+    from dolfin import __version__ as dolfin_version
+    from dolfin import has_linear_algebra_backend, has_slepc
+except ImportError:
+    raise
+else:
+    fenics_version_major = int(dolfin_version[0:4])
+    if (fenics_version_major < 2016 or fenics_version_major >= 2018):
+        raise Exception(
+            'hIPPYlib requires FEniCS versions 2016.x.x or 2017.x.x')
+    if not has_linear_algebra_backend("PETSc"):
+        raise Exception(
+            'hIPPYlib requires FEniCS to be installed with PETSc support')
+    if not has_slepc():
+        raise Exception(
+            'hIPPYlib requires FEniCS to be installed with SLEPc support')
+    
+
+here = path.abspath(path.dirname(__file__))
+with open(path.join(here, 'README.md'), encoding='utf-8') as f:
+    long_description = f.read()
+
+version = {}
+with open(path.join(here, 'hippylib/version.py')) as f:
+    exec(f.read(), version)
+
+VERSION = version['__version__']
+
+REQUIREMENTS = [
+    'mpi4py',
+    'numpy',
+    'matplotlib',
+    'scipy'
+]
+
+EXTRAS = {
```

Comment:
@eldarkh ,

We should add another EXTRA target called `doc` that installs `sphinx`, `m2r`, `sphinx_rtd_theme`.
Also would it make sense to call the `EXTRA` target `notebooks` or `tutorials` instead of `jupyter`?

Reply:
But I excluded the entire doc folder, do you want it to be distributed to the user as well (I don't know if it is needed)? Yes, we can call `EXTRA` target whatever we want, `notebooks` sounds good to me

Reply:
@eldarkh ,

You are 100% right, we should not ship the `doc` folder with `pip`.
Can you change the name of the `jupyter` `extra` to `notebook`, both here and in `INSTALL.md`? (I would do it myself, but I'm not sure whether I should change the key or the value...)

Reply:
> But I excluded the entire doc folder, do you want it to be distributed to the user as well (I don't know if it is needed.

It's often requested for packaging in debian/linux. Many packager really like to have `.tar.gz ~= git checkout`... But i agree we can skip and add later if necessary.